src/main/java/VASSAL/build/module/map/KeyBufferer.java

```diff
@@ -100,8 +109,12 @@ public void mousePressed(MouseEvent e) {
     // Don't clear the buffer until we find the clicked-on piece
     // Because selecting a piece affects its visibility
     EventFilter filter = null;
+    BandSelectType bandSelect = BandSelectType.NONE;
     if (p != null) {
-      filter = (EventFilter) p.getProperty(Properties.SELECT_EVENT_FILTER);
+      filter = (EventFilter) p.getProperty(Properties.SELECT_EVENT_FILTER);      
+      if (!e.isPopupTrigger() && (boolean)p.getProperty(Properties.NON_MOVABLE)) {
```

Comment:
I got an NPE on this. `getProperty()` returns `Object`s. In this case, the object you get will be a `Boolean`, which will be unboxed to a `boolean`... unless you get a `null`, which throws an NPE when you try casting it to a non-object type. The thing you need to do here is `Boolean.TRUE.equals(p.getProperty(Properties.NON_MOVABLE))`.

Reply:
Ah! Thx. Fixed & checked in.